src/lib.rs

```diff
@@ -178,7 +179,7 @@ impl Drop for Worker {
         let count = self.inner.count.fetch_sub(1, Ordering::Relaxed);
         // We are the last worker
         if count == 1 {
-            if let Some(waker) = self.inner.waker.lock().unwrap().take() {
```

Comment:
I think this can be replaced with just `self.inner.waker.wake()`

Reply:
> I think this can be replaced with just `self.inner.waker.wake()`

cc @cfeitong

Reply:
Oh I thought you were the same person 😄 I'll just merge this and make the change myself.